client/components/search/index.jsx

```diff
@@ -255,13 +262,13 @@ module.exports = React.createClass( {
 						? this._keyListener.bind( this, 'openSearch' )
 						: null
 					}
-					aria-controls={ 'search-component-' + this.id }
+					aria-controls={ 'search-component-' + this.state.instanceId }
```

Comment:
Is this causing any problems with react tree reconciliation? As far as I remember, it turned out not to be a problem (though I'm not sure why not).